java/vector/src/main/java/org/apache/arrow/vector/BitVector.java

```diff
@@ -43,7 +43,7 @@
   private final Mutator mutator = new Mutator();
 
   int valueCount;
-  private int allocationSizeInBytes = INITIAL_VALUE_ALLOCATION;
+  private int allocationSizeInBytes = getSizeFromCount(INITIAL_VALUE_ALLOCATION);
```

Comment:
Do we want to fix `reset()` as well?

Reply:
Yes done. Thanks @icexelloss  for pointing out.

Reply:
LGTM +1